lib/storekit/error.rb

```diff
@@ -25,5 +25,41 @@ def initialize(status_code)
     def server_error?
       [21004, 21007, 21008].include?(status_code) || message == 'Unknown error'
     end
+
+    def self.new status_code
+      return super if self != ValidationError
+
+      @status_code = status_code
+      type =
+      case status_code
```

Comment:
Can you follow the indentation style and case statement style in the `initialize` method?